recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -49,35 +82,88 @@ def configure(self):
             except Exception:
                 pass
 
+        if not self.options.with_otlp:
+            try:
+                del self.options.with_otlp_grpc
+            except Exception:
+                pass
+
+            try:
+                del self.options.with_otlp_http
+            except Exception:
+                pass
+
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("abseil/20220623.0")
-        self.requires("grpc/1.50.1")
-        self.requires("libcurl/7.86.0")
-        self.requires("nlohmann_json/3.11.2")
-        self.requires("openssl/1.1.1s")
-        if Version(self.version) <= "1.4.1":
-            self.requires("opentelemetry-proto/0.11.0")
-        else:
-            self.requires("opentelemetry-proto/0.19.0")
-        self.requires("protobuf/3.21.4")
-        self.requires("thrift/0.17.0")
-        if Version(self.version) >= "1.3.0":
-            self.requires("boost/1.80.0")
+        if self.options.with_gsl:
+            self.requires("ms-gsl/4.0.0")
+
+        if self.options.with_abseil:
+            self.requires("abseil/20220623.0")
+
+        if self.options.with_otlp:
+            self.requires("protobuf/3.21.4")
+            if Version(self.version) <= "1.4.1":
+                self.requires("opentelemetry-proto/0.11.0")
+            else:
+                self.requires("opentelemetry-proto/0.19.0")
+
+            if self.options.with_otlp_grpc:
+                self.requires("grpc/1.50.1")
+
+        if (self.options.with_zipkin or
+           self.options.with_elasticsearch or
+           self.options.with_otlp_http or
+           self.options.with_etw):
+                self.requires("nlohmann_json/3.11.2")
+                self.requires("openssl/1.1.1s")
+
+        if (self.options.with_zipkin or
+           self.options.with_elasticsearch or
+           self.options.with_otlp_http):
+                self.requires("libcurl/7.87.0")
+
+        if self.options.with_prometheus:
+            self.requires("prometheus-cpp/1.1.0")
+
+        if self.options.with_jaeger:
+            self.requires("thrift/0.17.0")
+
+            if Version(self.version) >= "1.3.0":
+                self.requires("boost/1.81.0")
+
+    @property
+    def _required_boost_components(self):
+        return ["locale"] if self.options.with_jaeger and Version(self.version) >= "1.3.0" else []
 
     def validate(self):
-        if self.info.settings.compiler.cppstd:
+        if self.settings.get_safe("compiler.cppstd"):
             check_min_cppstd(self, self._minimum_cpp_standard)
         check_min_vs(self, "192")
 
         if self.settings.os != "Linux" and self.options.shared:
             raise ConanInvalidConfiguration(f"{self.ref} supports building shared libraries only on Linux")
 
+        if self.options.with_otlp_grpc and not self.options.with_otlp:
```

Comment:
Since there are deleted its preferred yo use the get safe access

Same for the next one